libr/io/p/io_gprobe.c

```diff
@@ -711,6 +711,8 @@ static int gprobe_read (struct gport *port, ut32 addr, ut8 *buf, ut32 count) {
 	int res;
 
 	if (!request || !reply) {
+		r_buf_free (request);
```

Comment:
This is error-prone, since you free both pointers even if one of them can be NULL.